app/backend/test/cooperation/cooperation_commands_test.clj

```diff
@@ -60,4 +63,69 @@
             (tu/local-command :cooperation/create-application
                               {:thk.project/id project-id
                                :cooperation.3rd-party/name third-party-name
-                               :application invalid-date-application}))))))
+                               :application invalid-date-application}))))
+
+    (testing "Application and project need to match when editing"
+      (is (thrown?
```

Comment:
consider using `is-thrown-with-data?` in test utils that can check for the ex-info